cluster/images/crossplane/Makefile

```diff
@@ -1,31 +1,36 @@
 # ====================================================================================
 # Setup Project
 
-PLATFORMS := linux_amd64 linux_arm64 linux_arm linux_ppc64le
 include ../../../build/makelib/common.mk
 
 # ====================================================================================
 #  Options
-IMAGE = $(BUILD_REGISTRY)/crossplane-$(ARCH)
-OSBASEIMAGE = gcr.io/distroless/static@sha256:d2b0ec3141031720cf5eedef3493b8e129bc91935a43b50562fbe5429878d96b
-include ../../../build/makelib/image.mk
+
+include ../../../build/makelib/imagelight.mk
 
 # ====================================================================================
 # Targets
 
-ifeq ($(PLATFORM),$(filter $(PLATFORM),darwin_amd64 darwin_arm64 windows_amd64)) 
-$(info Skipping image build for $(PLATFORM))
-img.build:
-else
 img.build:
 	@$(INFO) docker build $(IMAGE)
+	@$(MAKE) BUILD_ARGS="--load" img.build.shared
+	@$(OK) docker build $(IMAGE)
+
+img.publish:
+	@$(INFO) docker publish $(IMAGE)
+	@$(MAKE) BUILD_ARGS="--push" img.build.shared
+	@$(OK) docker publish $(IMAGE)
+
+img.build.shared:
 	@cp Dockerfile $(IMAGE_TEMP_DIR) || $(FAIL)
-	@cp $(OUTPUT_DIR)/bin/$(OS)_$(ARCH)/crossplane $(IMAGE_TEMP_DIR) || $(FAIL)
+	@cp -r $(OUTPUT_DIR)/bin/ $(IMAGE_TEMP_DIR)/bin || $(FAIL)
```

Comment:
FWIW, I think we do that so that not the whole folder is copied as context for build but if I remember correctly `buildx` doesn't do that anymore so we may not need to have that step.

Reply:
@muvaf for the way publish is setup here, we need all the binaries in the temp dir, but we only add the relevant one when building: https://github.com/crossplane/crossplane/pull/2718/files#diff-4fffe7c1e9a74b7f3dea7bf63fbb3e853c841d6067b453a2edb15449f14f2652R6